providers/CENNZWalletProvider.tsx

```diff
@@ -96,33 +101,42 @@ export default function CENNZWalletProvider({
 		if (!matchedAccount) return selectAccount(accounts[0]);
 
 		selectAccount(matchedAccount);
-	}, [wallet, accounts, selectAccount]);
+	}, [wallet, accounts, selectAccount, selectedWallet]);
 
 	//3. Fetch asset balance
 	const updateBalances = useCallback(async () => {
```

Comment:
should move this to `WalletProvider` and rely entirely on the `useSelectedAccount` to fetch the balance, would be cleaner

Reply:
moved to custom hook `useUpdateCENNZBalances`, not sure if it was worth it